Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -43,7 +45,12 @@ def ensure_running(self):
         with self._lock:
             if self._pid is not None:
                 # semaphore tracker was launched before, is it still running?
-                pid, status = os.waitpid(self._pid, os.WNOHANG)
+                try:
+                    pid, _ = os.waitpid(self._pid, os.WNOHANG)
+                except ChildProcessError:
+                    # The process must be death (no process with pid self._pid).
+                    pid = self._pid
```

Comment:
I don't think that it's useful to assign pid. If you you get the error: we already know that the process is dead.

Just put the "if not pid: ..." instead the try block, or maybe add an "else:" block.